test/facets/third-party-facet.spec.ts

```diff
@@ -0,0 +1,332 @@
+        const signature3 = await sign(unaccusedArchaeologist, sarcoId, "bytes32");
+        const arweaveArchSig = await sign(arweaveAchaeologist, arweavetxId, "string");
+
+
+        await embalmerFacet.connect(embalmer).finalizeSarcophagus(
+            sarcoId,
+            [
+                { account: archaeologist1.address, ...signature1 },
+                { account: archaeologist2.address, ...signature2 },
+                { account: unaccusedArchaeologist.address, ...signature3 },
+            ],
+            arweaveArchSig,
+            arweavetxId,
+        );
+    }
+
+    const _initialiseEnvironment = async () => {
+        const signers = await ethers.getSigners();
+
+        thirdParty = signers[0];
+        embalmer = signers[1];
+        paymentAccount = signers[2];
+        archaeologist1 = signers[3];
+        archaeologist2 = signers[4];
+        arweaveAchaeologist = signers[5];
+        receiverAddress = signers[6];
+        unaccusedArchaeologist = signers[7];
+
+        ({ diamondAddress, sarcoToken } = await deployDiamond());
+
+        await _distributeTokens();
+
+        // Approve signers on the sarco token so transferFrom will work
+        await _approveSpending();
+
+        thirdPartyFacet = await ethers.getContractAt("ThirdPartyFacet", diamondAddress);
+
+        // Setup archaeologists - add free bonds to their accounts
+        await _setupArcheologists();
+
+        // Create a sarcophagus with archaeologists assigned, with a 1 day resurrection time
+        await _setupTestSarcophagus();
+    }
+
+    describe("clean()", () => {
+        beforeEach(_initialiseEnvironment);
+
+        it("Should distribute sum of cursed bonds of bad-acting archaeologists to embalmer and the address specified by cleaner", async () => {
+            // Increasing by this much so that the sarco is definitely expired
+            await time.increase(time.duration.years(sarcoResurrectionTimeInDays));
+
+            let embalmerBalanceBefore = await sarcoToken.balanceOf(embalmer.address);
+            let paymentDestinationBalance = await sarcoToken.balanceOf(paymentAccount.address);
+
+            // before cleaning...
+            expect(paymentDestinationBalance).to.eq(0);
+
+            const tx = await thirdPartyFacet.connect(thirdParty).clean(sarcoId, paymentAccount.address);
+            const receipt = await tx.wait();
+
+            expect(receipt.status).to.equal(1);
+
+            let embalmerBalanceAfter = await sarcoToken.balanceOf(embalmer.address);
+            paymentDestinationBalance = await sarcoToken.balanceOf(paymentAccount.address);
+
+            // after cleaning...
+            expect(embalmerBalanceAfter.gt(embalmerBalanceBefore)).to.be.true;
+            expect(paymentDestinationBalance.gt(0)).to.be.true;
+
+            //TODO: Actually calculate sum, and verify on exact amounts instead
+        });
+
+        it("Should emit CleanUpSarcophagus on successful cleanup", async () => {
```

Comment:
I know we talked about this in stand up. These event tests should probably be left out until we find a solution so we don't get failing tests.